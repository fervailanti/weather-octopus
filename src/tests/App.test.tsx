import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import App from '../App'

describe('App Component', () => {
  it('Should render correctly', () => {
    const app = render(<App />)
    expect(app.container).toMatchSnapshot()
  })
})
