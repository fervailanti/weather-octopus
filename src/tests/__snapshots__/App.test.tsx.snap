// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should render correctly 1`] = `
<div>
  <div
    class="flex flex-col items-center p-4"
  >
    <div
      class="flex flex-col items-center mt-2 lg:my-4"
    >
      <img
        alt="octopus"
        class="h-16 lg:h-20 transition duration-200 ease-in-out lg:cursor-pointer transform lg:hover:scale-110 active:opacity-50"
        data-testid="octopus-logo"
        src="octopus.png"
      />
      <h1
        class="text-4xl text-white font-hairline"
      >
        Weather
        <span
          class="font-semibold"
        >
          Octopus
        </span>
      </h1>
    </div>
    <div
      class="flex flex-col md:flex-row justify-around w-full"
    >
      <div
        class="flex flex-col items-center md:m-4 w-full"
      >
        <p
          class="text-white font-thin text-2xl my-3"
        >
          Your current location ๐
        </p>
        <div
          class="group shadow-xl rounded-lg w-full h-full max-w-screen-md m-2 p-4 lg:p-6 transition duration-200 ease-in-out lg:cursor-pointer transform lg:hover:scale-105 active:opacity-75 bg-gradient-to-r from-orange-900 to-blue-900"
        >
          <div
            class="invisible flex flex-col h-full"
          >
            <img
              alt="reload-icon"
              class="h-6 absolute right-0 mr-5 opacity-0 lg:group-hover:opacity-100 transition duration-200 ease-in-out"
              src="[object Object]"
            />
            <div
              class="flex justify-between w-full"
            >
              <div
                class="h-full flex flex-col justify-between"
              >
                <div
                  class="flex flex-col"
                >
                  <h2
                    class="text-white"
                  >
                    <span
                      class="font-semibold text-2xl"
                    >
                      Unknow
                    </span>
                    <span
                      class="text-lg"
                    >
                      , 
                      Unknow
                    </span>
                  </h2>
                  <p
                    class="text-white font-thin mt-2"
                  >
                    As of 
                    <span
                      class="font-normal"
                    >
                      00:00
                    </span>
                  </p>
                </div>
                <div
                  class="flex flex-col"
                >
                  <p
                    class="text-white text-6xl font-semibold"
                  >
                    0
                    ยบ
                  </p>
                  <p
                    class="text-white font-thin"
                  >
                    Unknow weather description
                  </p>
                </div>
              </div>
              <div
                class="flex flex-col items-center self-end"
              >
                <img
                  alt="weather-icon"
                  class="h-20"
                  src="[object Object]"
                />
                <p
                  class="text-white mt-4 font-thin text-xl"
                >
                  <span
                    class="font-semibold"
                  >
                    0
                    ยบ
                  </span>
                   /
                   
                  <span
                    class="font-semibold"
                  >
                    0
                    ยบ
                  </span>
                </p>
              </div>
            </div>
            <div
              class="flex justify-around h-full items-center flex-wrap mt-6 border-t"
            >
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="visible absolute right-1/2 top-1/2 -mr-10 -mt-10"
          >
            <div
              class="loader"
            >
              <div />
              <div />
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col items-center md:m-4 w-full"
      >
        <p
          class="text-white font-thin text-2xl my-3"
        >
          Pinned city ๐
        </p>
        <div
          class="group shadow-xl rounded-lg w-full h-full max-w-screen-md m-2 p-4 lg:p-6 transition duration-200 ease-in-out lg:cursor-pointer transform lg:hover:scale-105 active:opacity-75 bg-gradient-to-r from-red-900 to-teal-900"
          id="pinned-city-forecast"
        >
          <div
            class="invisible flex flex-col h-full"
          >
            <img
              alt="reload-icon"
              class="h-6 absolute right-0 mr-5 opacity-0 lg:group-hover:opacity-100 transition duration-200 ease-in-out"
              src="[object Object]"
            />
            <div
              class="flex justify-between w-full"
            >
              <div
                class="h-full flex flex-col justify-between"
              >
                <div
                  class="flex flex-col"
                >
                  <h2
                    class="text-white"
                  >
                    <span
                      class="font-semibold text-2xl"
                    >
                      Unknow
                    </span>
                    <span
                      class="text-lg"
                    >
                      , 
                      Unknow
                    </span>
                  </h2>
                  <p
                    class="text-white font-thin mt-2"
                  >
                    As of 
                    <span
                      class="font-normal"
                    >
                      00:00
                    </span>
                  </p>
                </div>
                <div
                  class="flex flex-col"
                >
                  <p
                    class="text-white text-6xl font-semibold"
                  >
                    0
                    ยบ
                  </p>
                  <p
                    class="text-white font-thin"
                  >
                    Unknow weather description
                  </p>
                </div>
              </div>
              <div
                class="flex flex-col items-center self-end"
              >
                <img
                  alt="weather-icon"
                  class="h-20"
                  src="[object Object]"
                />
                <p
                  class="text-white mt-4 font-thin text-xl"
                >
                  <span
                    class="font-semibold"
                  >
                    0
                    ยบ
                  </span>
                   /
                   
                  <span
                    class="font-semibold"
                  >
                    0
                    ยบ
                  </span>
                </p>
              </div>
            </div>
            <div
              class="flex justify-around h-full items-center flex-wrap mt-6 border-t"
            >
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
              <div
                class="flex items-center mt-4 mr-4"
              >
                <img
                  alt="weather-icon"
                  class="h-6 mr-2"
                  src="[object Object]"
                />
                <div>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    Unknow dailys
                  </p>
                  <p
                    class="text-white font-thin text-sm"
                  >
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ /
                     
                    <span
                      class="font-normal"
                    >
                      0
                    </span>
                    ยบ
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="visible absolute right-1/2 top-1/2 -mr-10 -mt-10"
          >
            <div
              class="loader"
            >
              <div />
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col items-center md:mt-4 xl:mt-8"
    >
      <p
        class="text-white font-thin text-xl my-3"
      >
        Other cities you may also like:
      </p>
      <div
        class="flex md:flex-row flex-wrap justify-center mt-0"
      >
        <div
          class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
        >
          <img
            alt="alt"
            class="object-cover w-full h-24"
            src="new_york.jpg"
          />
          <div
            class="p-3"
          >
            <h2
              class="text-gray-900"
            >
              <span
                class="font-semibold text-xl"
              >
                New York
              </span>
              <span
                class="text-sm"
              >
                , 
                USA
              </span>
            </h2>
          </div>
        </div>
        <div
          class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
        >
          <img
            alt="alt"
            class="object-cover w-full h-24"
            src="paris.jpg"
          />
          <div
            class="p-3"
          >
            <h2
              class="text-gray-900"
            >
              <span
                class="font-semibold text-xl"
              >
                Paris
              </span>
              <span
                class="text-sm"
              >
                , 
                France
              </span>
            </h2>
          </div>
        </div>
        <div
          class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
        >
          <img
            alt="alt"
            class="object-cover w-full h-24"
            src="rome.jpg"
          />
          <div
            class="p-3"
          >
            <h2
              class="text-gray-900"
            >
              <span
                class="font-semibold text-xl"
              >
                Rome
              </span>
              <span
                class="text-sm"
              >
                , 
                Italy
              </span>
            </h2>
          </div>
        </div>
        <div
          class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
        >
          <img
            alt="alt"
            class="object-cover w-full h-24"
            src="tokyo.jpg"
          />
          <div
            class="p-3"
          >
            <h2
              class="text-gray-900"
            >
              <span
                class="font-semibold text-xl"
              >
                Tokyo
              </span>
              <span
                class="text-sm"
              >
                , 
                Japan
              </span>
            </h2>
          </div>
        </div>
        <div
          class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
        >
          <img
            alt="alt"
            class="object-cover w-full h-24"
            src="berlin.jpg"
          />
          <div
            class="p-3"
          >
            <h2
              class="text-gray-900"
            >
              <span
                class="font-semibold text-xl"
              >
                Berlin
              </span>
              <span
                class="text-sm"
              >
                , 
                Germany
              </span>
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
